crates/scroll/alloy/rpc-types/src/transaction.rs

```diff
@@ -35,6 +35,8 @@ impl Transaction {
             // For l1 messages, we set the `gasPrice` field to 0 in rpc
             0
         } else {
+            // TODO: should we get the pool base fee in the case where the transaction is a pending
+            // transaction here?
```

Comment:
This is one of the points that I'm not sure about: should the base fee from the transaction pool be fetched and used here? 

For context, this type is used in the `RpcConverter` and builds a RPC transaction from a recovered tx and the scroll transaction info (which contains the L1 fee and the block context info). For now, afaiu, the `RpcConverter` uses empty block context for pending pool transaction (see [here](https://github.com/scroll-tech/reth/blob/db0946cc1d401dd8c8fc164ca4fa653ed8d8bcec/crates/rpc/rpc-types-compat/src/transaction.rs#L39)) but from what I understand in l2geth, the pending base fee is used to compute the `effective_gas_price` for pool transactions.

Reply:
Where is the original value (`tx_info.inner.base_fee`) taken from, the latest or pending block?

I don't think it's very important to consider `overhead` here, and this will just be a transient discrepancy.

Reply:
It's taken from [here](https://github.com/scroll-tech/reth/blob/6051c38c90372d7fdbb72469dd2d8f4bce4299ca/crates/rpc/rpc-convert/src/transaction.rs#L48) and with thus be `None`.